src/db.rs

```diff
@@ -659,6 +659,33 @@ impl DB {
         })
     }
 
+    pub fn list_cf<P: AsRef<Path>>(opts: &Options, path: P) -> Result<Vec<String>, Error> {
+        let path = path.as_ref();
+        let cpath = match CString::new(path.to_string_lossy().as_bytes()) {
+            Ok(c) => c,
+            Err(_) => {
+                return Err(Error::new("Failed to convert path to CString \
+                                       when opening DB."
+                                      .to_owned()))
+            }
+        };
+
+        let mut length = 0;
+        let vec;
+
+        unsafe {
+            let ptr = ffi_try!(ffi::rocksdb_list_column_families(opts.inner,
+                                                                 cpath.as_ptr() as *const _,
+                                                                 &mut length));
+
+            vec = Vec::from_raw_parts(ptr, length, length).iter().map(|&ptr| {
+                CString::from_raw(ptr).into_string().unwrap()
+            }).collect();
+        }
+        Ok(vec)
```

Comment:
You can have `Ok(vec)` be the last line in the unsafe block, and the function will return it there. No need to declare `vec` on line 674 before assigning it.